### PR TITLE
Switch LTTng tests to CTest

### PR DESCRIPTION
**Pull request by Robo210**
- Remove catch2 dependency
- Set up CTest for lttng tests
- Add a bit more error checking in the tests

The tests still don't actually run lttng to collect events, so the logging in the tests will all turn into no-opts. Crashes in the tests won't be detected unless lttng is run externally.

### CODE REVIEW

File: LTTng/CMakeLists.txt [hunk 1]
```diff
@@ -14,20 +14,5 @@ find_package(lttng-ust-tracepoint REQUIRED)
...
-            add_subdirectory("${PROJECT_SOURCE_DIR}/../external/Catch2" Catch2)
-        else ()
-            find_package(Catch2 REQUIRED)
-        endif ()
-
-        add_subdirectory(test)
-    endif ()
-endif ()
+# Use the standard BUILD_TESTING option to disable
+add_subdirectory(test)
```
Comment (idigdoug) [inline LTTng/CMakeLists.txt:18]: I think you need to do
```
if(BUILD_TESTING)
    add_subdirectory(test)
endif()
```

File: LTTng/test/TestTraceLogging.h [hunk 2]
```diff
@@ -1,23 +1,15 @@
...
 
 typedef struct Buffer
 {
     void* Buffer;
     uint16_t Length;
 } Buffer;
 
-static bool TestCommon(void)
+// Returns 0 on success
+static int TestCommon()
```
Comment (idigdoug) [inline LTTng/test/TestTraceLogging.h:12]: Keep the `(void)`. It's meaningful in C.

File: LTTng/test/main.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+extern "C" int TestC();
+int TestCpp();
```
Comment (idigdoug) [inline LTTng/test/main.cpp:2]: Please keep these in the header.

**Review by idigdoug — CHANGES_REQUESTED** [review]

File: LTTng/CMakeLists.txt (hunk 1 above)
Comment (Robo210) [reply LTTng/CMakeLists.txt:18]: We're always going to build the test subdir, there's no reason not to. BUILD_TESTING controls whether CTest calls the enable_testing() macro to actually hook up the tests into the makefile.

Comment (idigdoug) [reply LTTng/CMakeLists.txt]: It is desirable to allow users to not even build the tests. If the test has dependencies that the user doesn't want to figure out how to provide, the user shouldn't have to install them just to get the project to build. And if the test has some build failure like a C++17 dependency, they user should be able to disable building to skip the issue.

**Review by idigdoug — APPROVED** [review]

**Pull request merged**